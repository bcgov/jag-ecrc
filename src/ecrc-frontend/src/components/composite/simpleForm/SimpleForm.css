.simpleForm {
  width: 100%;
  overflow: hidden;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 10px;
}

.simpleForm_title {
  width: 100%;
  background-color: rgba(56, 89, 138, 1);
  overflow: hidden;
  padding-left: 20px;
  padding-top: 5px;
  padding-bottom: 5px;
  color: white;
  font-size: 18px;
  font-weight: 600;
  text-align: left;
  line-height: 22px;
  vertical-align: middle;
}

ul {
  margin: 0px;
  list-style-type: none;
  width: 100%;
}

ul li {
  margin: 10px;
}

ul#buttonList li {
  display: inline;
  padding: 5px;
}
