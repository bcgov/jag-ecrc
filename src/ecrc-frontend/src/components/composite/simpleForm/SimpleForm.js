import React from "react";
import PropTypes from "prop-types";
import "./SimpleForm.css";
import { TextInput } from "../../base/textInput/TextInput";
import { Button } from "../../base/button/Button";

export const SimpleForm = ({ simpleForm: { title, textInputs, buttons } }) => {
  const inputList = textInputs.map(input => {
    return (
      <li key={input.id}>
        <TextInput textInput={input} onChange={input.onChange} />
      </li>
    );
  });

  const buttonList = buttons.map(button => {
    return (
      <li key={button.label}>
        <Button
          button={{
            label: button.label,
            buttonStyle: button.buttonStyle,
            buttonSize: button.buttonSize,
            type: button.type
          }}
          onClick={button.onClick}
        />
      </li>
    );
  });

  return (
    <form className="simpleForm">
      {title !== null && <div className="simpleForm_title">{title}</div>}
      {inputList.length > 0 && <ul>{inputList}</ul>}
      {buttonList.length > 0 && <ul id="buttonList">{buttonList}</ul>}
    </form>
  );
};

SimpleForm.propTypes = {
  simpleForm: PropTypes.shape({
    title: PropTypes.string,
    textInputs: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        textInputStyle: PropTypes.string,
        value: PropTypes.string,
        onChange: PropTypes.func,
        isRequired: PropTypes.bool
      })
    ),
    buttons: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        buttonStyle: PropTypes.string.isRequired,
        buttonSize: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        onClick: PropTypes.func
      })
    )
  }).isRequired
};

export default SimpleForm;
