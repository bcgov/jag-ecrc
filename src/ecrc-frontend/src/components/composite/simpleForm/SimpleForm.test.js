/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { create } from "react-test-renderer";

import { SimpleForm } from "./SimpleForm";

describe("SimpleForm Component", () => {
  const textInputs = [
    {
      label: "label1",
      id: "textInput1",
      onChange: () => jest.fn()
    },
    {
      label: "label3",
      id: "textInput3",
      textInputStyle: "textinput_editable_gray",
      isRequired: true
    }
  ];

  const buttons = [
    {
      label: "button2",
      buttonStyle: "btn btn-warning",
      buttonSize: "btn btn-lg",
      type: "button",
      onClick: () => jest.fn()
    },
    {
      label: "button3",
      buttonStyle: "btn btn-danger",
      buttonSize: "btn btn-sm",
      type: "button",
      onClick: () => jest.fn()
    }
  ];

  const simpleForm = { title: "title", textInputs, buttons };

  test("Matches the snapshot", () => {
    const form = create(<SimpleForm simpleForm={simpleForm} />);
    expect(form.toJSON()).toMatchSnapshot();
  });
});
