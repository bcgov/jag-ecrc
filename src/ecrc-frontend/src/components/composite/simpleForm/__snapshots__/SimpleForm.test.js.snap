// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleForm Component Matches the snapshot 1`] = `
<form
  className="simpleForm"
>
  <div
    className="simpleForm_title"
  >
    title
  </div>
  <ul>
    <li>
      <div>
        <label
          className="label"
          htmlFor="textInput1"
        >
          <div
            className="textinput_label"
          >
            label1
          </div>
          
           
          <span
            className="note"
          />
        </label>
        <input
          className="textinput_editable_white"
          id="textInput1"
          onChange={[Function]}
          type="text"
        />
        <span
          className="error"
        />
      </div>
    </li>
    <li>
      <div>
        <label
          className="label"
          htmlFor="textInput3"
        >
          <div
            className="textinput_label"
          >
            label3
          </div>
          <span
            className="mandatory"
            id="asterisk"
          >
            *
          </span>
           
          <span
            className="note"
          />
        </label>
        <input
          className="textinput_editable_gray"
          id="textInput3"
          onChange={[Function]}
          type="text"
        />
        <span
          className="error"
        />
      </div>
    </li>
  </ul>
  <ul
    id="buttonList"
  >
    <li>
      <button
        className="btn btn-warning btn btn-lg"
        onClick={[Function]}
        type="button"
      >
        button2
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
    </li>
    <li>
      <button
        className="btn btn-danger btn btn-sm"
        onClick={[Function]}
        type="button"
      >
        button3
        <div
          className="loader-hide"
        >
          <div
            className="btn-loader"
          />
        </div>
      </button>
    </li>
  </ul>
</form>
`;
